src/model/version.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) 2022 spdx contributors
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import re
+from re import Pattern
+
+
+class Version:
+    VERSION_REGEX: Pattern = re.compile(r"^(\d+)\.(\d+)$")
+
+    major: int
+    minor: int
+
+    @classmethod
+    def is_valid_version_string(cls, value: str) -> bool:
```

Comment:
Because of the intended usage of `from_string`: I changed that to throw an error instead of returning `None` if the provided string is invalid. A caller should make sure a string is valid before calling `from_string` (this is just a change in how the control flow works. Previously, the caller would have to handle the `None` case separately, so there is no real difference in the number of cases to handle).

`from_string` should also use this method, btw, I'll change that quickly